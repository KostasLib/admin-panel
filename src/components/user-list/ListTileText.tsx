import React from 'react';
import cn from 'classnames';
import s from './ListTileText.module.scss';

interface IProps {
  primary: string;
  secondary?: string;
  selected?: boolean;
}

/** A function component that renders a user's details in a list tile. Collapses on mobile. */
export const ListTileText: React.FC<IProps> = props => {
  const { primary, secondary, selected } = props;

  // classnames
  const primaryClass = cn(s['primary'], { [s['primary--selected']]: selected });
  const secondaryClass = cn(s['secondary'], { [s['secondary--selected']]: selected });

  return (
    <div className={s['text-wrapper']}>
      <p className={primaryClass}>{primary}</p>
      {secondary && <p className={secondaryClass}>{secondary}</p>}
    </div>
  );
};
