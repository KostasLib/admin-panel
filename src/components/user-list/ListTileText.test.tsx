import React from 'react';
import { shallow } from 'enzyme';
import { ListTileText } from './ListTileText';

describe('The <ListTileText /> component', () => {
  it('renders primary and secondary texts, if both are provided', () => {
    const component = shallow(<ListTileText primary="Primary" secondary="Secondary" />);

    expect(component).toMatchSnapshot();
  });
  it('renders only the primary text, if the secondary is omitted', () => {
    const component = shallow(<ListTileText primary="Primary" />);

    expect(component).toMatchSnapshot();
  });
  it('applies the selected className modifier to each child if the selected prop is true', () => {
    const component = shallow(
      <ListTileText primary="Primary" secondary="Secondary" selected />,
    );
    expect(component).toMatchSnapshot();
  });
});
