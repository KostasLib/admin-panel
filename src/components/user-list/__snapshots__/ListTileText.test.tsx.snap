// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <ListTileText /> component applies the selected className modifier to each child if the selected prop is true 1`] = `
<div
  className="text-wrapper"
>
  <p
    className="primary primary--selected"
  >
    Primary
  </p>
  <p
    className="secondary secondary--selected"
  >
    Secondary
  </p>
</div>
`;

exports[`The <ListTileText /> component renders only the primary text, if the secondary is omitted 1`] = `
<div
  className="text-wrapper"
>
  <p
    className="primary"
  >
    Primary
  </p>
</div>
`;

exports[`The <ListTileText /> component renders primary and secondary texts, if both are provided 1`] = `
<div
  className="text-wrapper"
>
  <p
    className="primary"
  >
    Primary
  </p>
  <p
    className="secondary"
  >
    Secondary
  </p>
</div>
`;
